exercises-cpp/donald_chung/ex01_basics/src/main.cpp

```diff
@@ -0,0 +1,79 @@
+/**
+ * Skeleton main routine
+ */
+#include <algorithm>
+#include <string>
+#include <iostream>
+#include <map>
+#include <fstream>
+#include <vector>
+
+typedef std::map<std::string, int> word_counting_container; //typealias
+
+void word_counting(std::istream& input, word_counting_container& word_counts)
+{
+	std::string current_word;
+	while (input >> current_word)
+	{
+		current_word.erase(std::remove_if(current_word.begin(), current_word.end(), ispunct), current_word.end());//remove punctuation
+		std::for_each(current_word.begin(), current_word.end(), [](char& c) {
+			c = ::tolower(c);
+			});
+		if (current_word.length() > 4)
+		{
+			++word_counts[current_word];
+		}
+	}
+}
+
+bool compare(std::pair<std::string, int>& a, std::pair<std::string, int>& b)
+{
+	return a.second > b.second;
+}
+
+void sorting(word_counting_container& word_counts, std::vector<std::pair<std::string, int>>& arrange)
+{
+
+	for (auto& it : word_counts) {
+		arrange.emplace_back(it);
+	}
+	std::sort(arrange.begin(), arrange.end(), compare);
+}
+
+void output_result(std::vector<std::pair<std::string, int>>& arrange)
+{
+	std::ofstream outf{ "result.txt" };
```

Comment:
At the end of this function I think we want to close the output file stream with `outf.close();`